sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/WritableContent.java

```diff
@@ -0,0 +1,253 @@
+                throw new IOException("openForWrite(..) must be called before invoking writeBlock(..).");
+            }
+            // When `useContentResolver` is true then ContentUri must be treated as an opaque handle
+            // and the raw-path must not be used.
+            // https://commonsware.com/blog/2016/03/15/how-consume-content-uri.html
+            // Obtaining a RandomAccessFile requires the raw-path to the file backing the ContentUri.
+            //
+            // So to write to Content, instead of RandomAccessFile we will use FileChannel, specifically
+            // we a shared instance of FileChannel to write the blocks downloaded by concurrent (OkHttp) threads.
+            // Operations in FileChannel instance are concurrent safe.
+            // https://developer.android.com/reference/java/nio/channels/FileChannel
+            //
+            // The FileChannel instance is obtained from a FileOutputStream for the ContentUri
+            // (see WriteToContentChannel). Only one instance of FileOutputStream can be opened
+            // in "write-mode" so we share the FileOutputStream instance and it's backing
+            // FileChannel instance.
+            // https://docs.oracle.com/javase/7/docs/api/java/io/FileOutputStream.html
+            //
+            this.contentChannel.writeBlock(blockOffset, block);
+        } else {
+            try (RandomAccessFile randomAccessFile = new RandomAccessFile(this.contentUri.getPath(), "rw")) {
+                randomAccessFile.seek(blockOffset);
+                randomAccessFile.write(block);
+            }
+        }
+    }
+
+    /**
+     * close the content.
+     *
+     * @throws IOException if close operation fails
+     */
+    void close() throws IOException {
+        if (this.useContentResolver) {
+            synchronized (this) {
+                if (this.contentChannel != null) {
+                    this.contentChannel.close();
+                }
+            }
+        }
+    }
+
+    /**
+     * Check a persistable write permission is granted on the content.
+     *
+     * @param context the context to access content resolver
+     * @param contentUri the content uri
+     * @throws IllegalStateException if permission is not granted
+     */
+    private static void checkPersistableWriteGranted(Context context, Uri contentUri) throws IllegalStateException {
+        final List<UriPermission> permissions = context.getContentResolver().getPersistedUriPermissions();
+        boolean grantedWrite = false;
+        for (UriPermission permission : permissions) {
+            if (permission.isWritePermission()) {
+                grantedWrite = true;
+                break;
+            }
+        }
+
+        if (!grantedWrite) {
+            throw new IllegalStateException("Write permission for the content '" + contentUri + "' is not granted or revoked.");
+        }
+    }
+
+    /**
+     * A Channel to write to a Content identified by a ContentUri.
```

Comment:
Do we want to capitalize Content wherever its mentioned to make the distinction between ReadableContent/WritableContent and the word "content" in the general sense?

Reply:
"content" is a general android term so it should be fine to use it in comments as "content", this specific code comment, maybe I should change from "Content" to "content".